R/insertPackage.R

```diff
@@ -305,6 +308,11 @@ getPackageInfo <- function(file) {
 
     fields <- strsplit(builtstring, "; ")[[1]]
     names(fields) <- c("Rversion", "OSflavour", "Date", "OS")
+    
+    # Insert the OSflavour if specified by the user:
+    if(length(OSflavour) && is.character(OSflavour)) {
```

Comment:
If we set the default to `NULL` we should check for null.  Else instead of `NULL` maybe use `character()` ?  Then your test for length and type is good.

Reply:
I made the changes you suggested (character() instead of NULL as default, documentation of OSflavour wrapped to 100 characters with suggested value R.Version()$platform, and an entry in the ChangeLog).

Reply:
Thanks so much -- it's long been merged :)    And a casual `dratInsert.r` (using a wrapper from `littler`) worked fine for me too (and I generally have a vanilla setup with source tarballs only in my `drat`-using repos).